libcaf_core/caf/delegate.test.cpp

```diff
@@ -0,0 +1,109 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#include "caf/test/caf_test_main.hpp"
+#include "caf/test/fixture/deterministic.hpp"
+#include "caf/test/test.hpp"
+
+#include "caf/event_based_actor.hpp"
+#include "caf/scoped_actor.hpp"
+#include "caf/sec.hpp"
+#include "caf/send.hpp"
+
+namespace fixture = caf::test::fixture;
+
+WITH_FIXTURE(fixture::deterministic) {
+
+#ifdef CAF_ENABLE_EXCEPTIONS
```

Comment:
Why is this here?

Reply:
In the deterministic.test.cpp file all the tests using expect<> was written with exceptions enabled. We also need to add some `should_fail_with_exception` in the tests in future in my opinion. Shall I remove them and group the checks with `should_fail_with_exception` seperately?

Reply:
We won't need `should_fail_with_exception` here. That was for testing the fixture itself. If we need something that checks that a message is *not* present, we'll add another predicate. The old framework has `disallow` for this, so we'll probably bring that back.